include/ignition/math/detail/Box.hh

```diff
@@ -113,6 +121,159 @@ T Box<T>::Volume() const
   return this->size.X() * this->size.Y() * this->size.Z();
 }
 
+//////////////////////////////////////////////////
+template <typename T>
+std::vector<std::pair<Triangle3<T>, T>> TrianglesInPlane(
+    const Plane<T> &_plane, IntersectionPoints<T> &_vertices)
+{
+  std::vector<std::pair<Triangle3<T>, T>> triangles;
+  std::vector<Vector3<T>> pointsInPlane;
+
+  Vector3<T> centroid;
+  for (const auto &pt : _vertices)
+  {
+    if (_plane.Side(pt) == Plane<T>::NO_SIDE)
+    {
+      pointsInPlane.push_back(pt);
+      centroid += pt;
+    }
+  }
+  centroid /= T(pointsInPlane.size());
+
+  if (pointsInPlane.size() < 3)
+    return {};
+
+  // Choose a basis in the plane of the triangle
+  auto axis1 = (pointsInPlane[0] - centroid).Normalize();
+  auto axis2 = axis1.Cross(_plane.Normal()).Normalize();
+
+  // Since the polygon is always convex, we can try to create a fan of triangles
+  // by sorting the points by their angle in the plane basis.
+  std::sort(pointsInPlane.begin(), pointsInPlane.end(),
+    [centroid, axis1, axis2] (const Vector3<T> &_a, const Vector3<T> &_b)
+    {
+      auto aDisplacement = _a - centroid;
+      auto bDisplacement = _b - centroid;
+
+      // project line onto the new basis vectors
+      // The axis length will never be zero as we have three different points
+      // and the centroid will be different.
+      auto aX = axis1.Dot(aDisplacement) / axis1.Length();
+      auto aY = axis2.Dot(aDisplacement) / axis2.Length();
+
+      auto bX = axis1.Dot(bDisplacement) / axis1.Length();
+      auto bY = axis2.Dot(bDisplacement) / axis2.Length();
+
+      return atan2(aY, aX) < atan2(bY, bX);
+    });
+  for (std::size_t i = 0; i < pointsInPlane.size(); ++i)
+  {
+    triangles.emplace_back(
+      Triangle3<T>(pointsInPlane[i],
+        pointsInPlane[(i + 1) % pointsInPlane.size()], centroid),
+      (_plane.Side({0, 0, 0}) == Plane<T>::POSITIVE_SIDE) ? -1 : 1);
+  }
+
+  return triangles;
+}
+
+/////////////////////////////////////////////////
+template<typename T>
+T Box<T>::VolumeBelow(const Plane<T> &_plane) const
+{
+  auto verticesBelow = this->VerticesBelow(_plane);
+  if (verticesBelow.empty())
+    return 0;
+
+  auto intersections = this->Intersections(_plane);
+  verticesBelow.merge(intersections);
```

Comment:
I believe this implicitly uses the `==` operator for `Vector3`, which has a default tolerance of `1e-6`. If tolerances become an issue, we could overload this function with an `_epsilon` parameter in the future, but I don't think that is necessary now

Reply:
Added a todo in f547e71.